pennylane/devices/default_tensor.py

```diff
@@ -656,9 +663,24 @@ def simulate(self, circuit: QuantumScript) -> Result:
         # The state is reset every time a new circuit is executed, and number of wires
         # is established at runtime to match the circuit if not provided.
         wires = circuit.wires if self.wires is None else self.wires
-        self._quimb_circuit = self._initial_quimb_circuit(wires)
+        operations = copy.deepcopy(circuit.operations)
```

Comment:
Now we don't need the `deepcopy` method anymore (we can simply use the `copy` method in `Circuit`). Probably it can also be replaced here

Reply:
This is not a copy of the underlying Circuit object but of the tape's operations list.